<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TransactionsFrame</class>
 <widget class="QFrame" name="TransactionsFrame">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>894</width>
    <height>638</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Frame</string>
  </property>
  <property name="frameShape">
   <enum>QFrame::NoFrame</enum>
  </property>
  <property name="frameShadow">
   <enum>QFrame::Plain</enum>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>25</number>
   </property>
   <property name="topMargin">
    <number>23</number>
   </property>
   <property name="rightMargin">
    <number>25</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="spacing">
      <number>10</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="WalletGui::WalletHeaderLabel" name="m_transactionsTextLabel">
       <property name="text">
        <string>Transactions</string>
       </property>
       <property name="indent">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_4">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="WalletGui::WalletTinyLinkLikeButton" name="m_resetFilterButton">
       <property name="cursor">
        <cursorShape>PointingHandCursor</cursorShape>
       </property>
       <property name="text">
        <string>Reset</string>
       </property>
       <property name="flat">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="WalletGui::WalletTinyLinkLikeButton" name="m_filterButton">
       <property name="cursor">
        <cursorShape>PointingHandCursor</cursorShape>
       </property>
       <property name="focusPolicy">
        <enum>Qt::NoFocus</enum>
       </property>
       <property name="text">
        <string>Show filter</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="flat">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer_3">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>10</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QFrame" name="m_filterFrame">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <property name="spacing">
       <number>25</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>5</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>30</number>
      </property>
      <item>
       <widget class="QFrame" name="m_filterPeriodComboFrame">
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <property name="spacing">
          <number>8</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="WalletGui::WalletNormalGrayTextLabel" name="m_filterTextLabel">
           <property name="text">
            <string>PERIOD</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="m_filterCombo">
           <property name="focusPolicy">
            <enum>Qt::NoFocus</enum>
           </property>
           <item>
            <property name="text">
             <string>All time</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Last hour</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Last day</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Last week</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Last month</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Last year</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Custom</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="m_filterPeriodFrame">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <property name="spacing">
          <number>8</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="WalletGui::WalletNormalGrayTextLabel" name="m_filterBeginTextLabel">
           <property name="text">
            <string>FROM</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDateTimeEdit" name="m_filterBeginDtedit">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="displayFormat">
            <string>dd MMM yyyy, HH:mm</string>
           </property>
           <property name="calendarPopup">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="WalletGui::WalletNormalGrayTextLabel" name="m_filterEndTextLabel">
           <property name="text">
            <string>TO</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDateTimeEdit" name="m_filterEndDtedit">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="displayFormat">
            <string>dd MMM yyyy, hh:mm</string>
           </property>
           <property name="calendarPopup">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="m_filterHashFrame">
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <property name="spacing">
          <number>8</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="WalletGui::WalletNormalGrayTextLabel" name="m_filterHashTextLabel">
           <property name="text">
            <string>HASH</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="m_filterHashEdit"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="m_filterAddressFrame">
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <property name="spacing">
          <number>8</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="WalletGui::WalletNormalGrayTextLabel" name="m_filterAddressTextLabel">
           <property name="text">
            <string>ADDRESS</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="m_filterAddressEdit"/>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>10</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="WalletGui::WalletTableView" name="m_transactionsView">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="focusPolicy">
      <enum>Qt::NoFocus</enum>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
     <property name="tabKeyNavigation">
      <bool>false</bool>
     </property>
     <property name="dragDropOverwriteMode">
      <bool>false</bool>
     </property>
     <property name="alternatingRowColors">
      <bool>true</bool>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::NoSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectItems</enum>
     </property>
     <property name="textElideMode">
      <enum>Qt::ElideMiddle</enum>
     </property>
     <property name="verticalScrollMode">
      <enum>QAbstractItemView::ScrollPerPixel</enum>
     </property>
     <property name="horizontalScrollMode">
      <enum>QAbstractItemView::ScrollPerPixel</enum>
     </property>
     <property name="showGrid">
      <bool>false</bool>
     </property>
     <property name="gridStyle">
      <enum>Qt::NoPen</enum>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
     <property name="cornerButtonEnabled">
      <bool>false</bool>
     </property>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>false</bool>
     </attribute>
     <attribute name="verticalHeaderVisible">
      <bool>false</bool>
     </attribute>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>10</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="WalletGui::WalletNormalBlueButton" name="m_exportToCsvButton">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>110</width>
         <height>30</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>110</width>
         <height>30</height>
        </size>
       </property>
       <property name="focusPolicy">
        <enum>Qt::NoFocus</enum>
       </property>
       <property name="text">
        <string>Save as CSV</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>WalletGui::WalletNormalBlueButton</class>
   <extends>QPushButton</extends>
   <header>Gui/Common/WalletBlueButton.h</header>
  </customwidget>
  <customwidget>
   <class>WalletGui::WalletHeaderLabel</class>
   <extends>QLabel</extends>
   <header>Gui/Common/WalletTextLabel.h</header>
  </customwidget>
  <customwidget>
   <class>WalletGui::WalletNormalGrayTextLabel</class>
   <extends>QLabel</extends>
   <header>Gui/Common/WalletTextLabel.h</header>
  </customwidget>
  <customwidget>
   <class>WalletGui::WalletTinyLinkLikeButton</class>
   <extends>QPushButton</extends>
   <header>Gui/Common/WalletLinkLikeButton.h</header>
  </customwidget>
  <customwidget>
   <class>WalletGui::WalletTableView</class>
   <extends>QTableView</extends>
   <header>Gui/Common/WalletTableView.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>m_exportToCsvButton</sender>
   <signal>clicked()</signal>
   <receiver>TransactionsFrame</receiver>
   <slot>exportToCsv()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>761</x>
     <y>587</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_filterCombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>TransactionsFrame</receiver>
   <slot>filterChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>93</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_filterEndDtedit</sender>
   <signal>dateTimeChanged(QDateTime)</signal>
   <receiver>TransactionsFrame</receiver>
   <slot>filterPeriodChanged(QDateTime)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>243</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_filterHashEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>TransactionsFrame</receiver>
   <slot>filterHashChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>566</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_filterAddressEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>TransactionsFrame</receiver>
   <slot>filterAddressChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>566</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_filterBeginDtedit</sender>
   <signal>dateTimeChanged(QDateTime)</signal>
   <receiver>TransactionsFrame</receiver>
   <slot>filterPeriodChanged(QDateTime)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>93</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_filterButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>TransactionsFrame</receiver>
   <slot>showFilter(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>38</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_resetFilterButton</sender>
   <signal>clicked()</signal>
   <receiver>TransactionsFrame</receiver>
   <slot>resetClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>712</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>446</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_filterButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_resetFilterButton</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>814</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>712</x>
     <y>39</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_transactionsView</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>TransactionsFrame</receiver>
   <slot>transactionDoubleClicked(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>446</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>446</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_transactionsView</sender>
   <signal>clicked(QModelIndex)</signal>
   <receiver>TransactionsFrame</receiver>
   <slot>transactionClicked(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>446</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>446</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>exportToCsv()</slot>
  <slot>transactionDoubleClicked(QModelIndex)</slot>
  <slot>filterChanged(int)</slot>
  <slot>filterPeriodChanged(QDateTime)</slot>
  <slot>filterHashChanged(QString)</slot>
  <slot>filterAddressChanged(QString)</slot>
  <slot>showFilter(bool)</slot>
  <slot>resetClicked()</slot>
  <slot>transactionClicked(QModelIndex)</slot>
 </slots>
</ui>
